Added flags to configure preamble

alexandervanrenen opened the pull request:
Hi,

I was looking for a logging lib and found this one. It looks great (thanks) and it almost did everything I wanted. But there was no option to configure the preamble (it is a bit to verbose for my purposes), so I added this feature.

Feel free to reject if you don't like it ;)

\- Alex

emilk commented on loguru.hpp:
```diff
@@ -2301,6 +2319,30 @@ namespace loguru
+	{
+		long pos = 0;
+		snprintf(out_buff, out_buff_size, ""); // Make sure there is a '\0' and handle out_buff_size==0
+ 		if(g_preamble_date) {
```
Please follow the Loguru code style with a space after if (`if (g_preamble_date) {`) here and elsewhere

emilk commented:
This looks very nice! I'm gonna take it for a spin to make sure it doesn't degrade performance.

emilk commented:
This PR causes a minor performance regression with about 10% in buffered mode.

One fix is to use the old code if everything in the preamble is turned on, but I'd rather keep the code readable in this case. If people want the highest possible speed they should be using either `RAW_LOG_F` (which is not slowed down by this PR) – or use another logging library. In other words: I feel the added benefit of this PR out-weights the small performance loss.

emilk commented:
A smaller concern is how the last `| ` is also controlled by `g_preamble_verbose`.

Example:

`g_preamble_verbose = true`:
```
[ thread name/id ]                   file:line     v| 
[main thread     ]      some_big_file.hpp:12345 WARN| Some warning
```

`g_preamble_verbose = false`:
```
[ thread name/id ]                   file:line     
[main thread     ]      some_big_file.hpp:12345 Some warning
```

If we instead control the last `| ` with a separate flag (`g_preamble_pipe`?) we could have this as:
```
[ thread name/id ]                   file:line  |    
[main thread     ]      some_big_file.hpp:12345 | Some warning
```

This is arguably more readable. Thoughts?

emilk commented on loguru.hpp:
```diff
@@ -2301,6 +2319,30 @@ namespace loguru
+ 		}
+ 		if(g_preamble_verbose) {
+ 			pos += snprintf(out_buff + pos, out_buff_size - pos, "v| ");
+ 		}
```
These new lines have a space before the first tab:

<img width="694" alt="screen shot 2018-01-21 at 21 03 05" src="https://user-images.githubusercontent.com/1148717/35198394-8b7c7f2e-feee-11e7-80a9-328fa84e6bf0.png">

alexandervanrenen commented:
Hi,

Thanks for your feedback.

I fixed the style issues (sorry about these). And I like the idea about controlling the pipe separately. This should all work now. I also fixed a small issue with the spaces of the file column. This should be correct now ;)

About the performance: Whatever you prefer.

- Alex

emilk commented:
Awesome, thanks!
